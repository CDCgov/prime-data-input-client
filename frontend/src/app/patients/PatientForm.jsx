import React, { useState } from "react";
//import { gql, useQuery, useMutation } from "@apollo/client";
import { gql, useMutation } from "@apollo/client";
import { toast } from "react-toastify";
import {
  PATIENT_TERM_PLURAL_CAP,
  PATIENT_TERM_CAP,
  stateCodes,
} from "../../config/constants";
import Breadcrumbs from "../commonComponents/Breadcrumbs";
import TextInput from "../commonComponents/TextInput";
import RadioGroup from "../commonComponents/RadioGroup";
import Checkboxes from "../commonComponents/Checkboxes";
import { Prompt } from "react-router-dom";
import moment from "moment";
import Dropdown from "../commonComponents/Dropdown";
import { displayFullName, showNotification } from "../utils";
import "./EditPatient.scss";
import Alert from "../commonComponents/Alert";

const ADD_PATIENT = gql`
  mutation(
    $lookupId: String
    $firstName: String!
    $middleName: String
    $lastName: String!
    $birthDate: String!
    $street: String!
    $streetTwo: String
    $city: String
    $state: String!
    $zipCode: String!
    $telephone: String!
    $role: String
    $email: String
    $county: String
    $race: [String]
    $ethnicity: String
    $gender: String
    $residentCongregateSetting: Boolean!
    $employedInHealthcare: Boolean!
  ) {
    addPatient(
      lookupId: $lookupId
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      birthDate: $birthDate
      street: $street
      streetTwo: $streetTwo
      city: $city
      state: $state
      zipCode: $zipCode
      telephone: $telephone
      role: $role
      email: $email
      county: $county
      race: $race
      ethnicity: $ethnicity
      gender: $gender
      residentCongregateSetting: $residentCongregateSetting
      employedInHealthcare: $employedInHealthcare
    )
  }
`;

const UPDATE_PATIENT = gql`
  mutation(
    $patientId: String!
    $lookupId: String
    $firstName: String!
    $middleName: String
    $lastName: String!
    $birthDate: String!
    $street: String!
    $streetTwo: String
    $city: String
    $state: String!
    $zipCode: String!
    $telephone: String!
    $role: String
    $email: String
    $county: String
    $race: [String]
    $ethnicity: String
    $gender: String
    $residentCongregateSetting: Boolean!
    $employedInHealthcare: Boolean!
  ) {
    updatePatient(
      patientId: $patientId
      lookupId: $lookupId
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      birthDate: $birthDate
      street: $street
      streetTwo: $streetTwo
      city: $city
      state: $state
      zipCode: $zipCode
      telephone: $telephone
      role: $role
      email: $email
      county: $county
      race: $race
      ethnicity: $ethnicity
      gender: $gender
      residentCongregateSetting: $residentCongregateSetting
      employedInHealthcare: $employedInHealthcare
    )
  }
`;

const Fieldset = (props) => (
  <fieldset className="prime-fieldset">
    <legend>
      <h3>{props.legend}</h3>
    </legend>
    {props.children}
  </fieldset>
);

const PatientForm = (props) => {
  const [addPatient] = useMutation(ADD_PATIENT);
  const [updatePatient] = useMutation(UPDATE_PATIENT);
  const [formChanged, setFormChanged] = useState(false);
  const [patient, setPatient] = useState(props.patient);

  const onChange = (e) => {
    let value = e.target.value;
    if (e.target.type === "checkbox") {
      value = {
        ...patient[e.target.name],
        [e.target.value]: e.target.checked,
      };
    }
    setFormChanged(true);
    setPatient({ ...patient, [e.target.name]: value });
  };
  const fullName = displayFullName(
    patient.firstName,
    patient.middleName,
    patient.lastName
  );

  const savePatientData = () => {
    setFormChanged(false);
    const variables = {
      lookupId: patient.lookupId,
      firstName: patient.firstName,
      middleName: patient.middleName,
      lastName: patient.lastName,
      birthDate: patient.birthDate,
      street: patient.street,
      streetTwo: patient.streetTwo,
      city: patient.city,
      state: patient.state,
      zipCode: patient.zipCode,
      telephone: patient.telephone,
      role: patient.role,
      email: patient.email,
      county: patient.county,
      race: patient.race ? Object.keys(patient.race) : null,
      ethnicity: patient.ethnicity,
      gender: patient.gender,
      residentCongregateSetting: patient.residentCongregateSetting === "YES",
      employedInHealthcare: patient.employedInHealthcare === "YES",
    };
    if (props.patientId) {
      updatePatient({
        variables: {
          patientId: props.patientId,
          ...variables,
        },
      }).then(
        () =>
          showNotification(
            toast,
            <Alert
              type="success"
              title={`${PATIENT_TERM_CAP} Record Saved`}
              body="Information record has been updated."
            />
          ),
        (error) => {
          console.error(error);
          showNotification(
            toast,
            <Alert
              type="error"
              title={`${PATIENT_TERM_CAP} Data Error`}
              body="Please check for missing data or typos."
            />
          );
        }
      );
    } else {
      addPatient({ variables }).then(
        () =>
          showNotification(
            toast,
            <Alert
              type="success"
              title={`${PATIENT_TERM_CAP} Record Created`}
              body="New information record has been created."
            />
          ),
        (error) => {
          console.error(error);
          showNotification(
            toast,
            <Alert
              type="error"
              title={`${PATIENT_TERM_CAP} Data Error`}
              body="Please check for missing data or typos."
            />
          );
        }
      );
    }
  };
  //TODO: when to save initial data? What if name isn't filled? required fields?
  return (
    <main className="prime-edit-patient prime-home">
      <Prompt
        when={formChanged}
        message={(location) =>
          "\nYour changes are not yet saved!\n\nClick OK discard changes, Cancel to continue editing."
        }
      />
      <Breadcrumbs
        crumbs={[
          { link: "../patients", text: PATIENT_TERM_PLURAL_CAP },
          {
            text: !props.patientId
              ? `Create New ${PATIENT_TERM_CAP}`
              : fullName,
          },
        ]}
      />
      <div className="prime-edit-patient-heading">
        <h2>
          {!props.patientId ? `Create New ${PATIENT_TERM_CAP}` : fullName}
        </h2>
        <button
          className="usa-button prime-save-patient-changes"
          disabled={!formChanged}
          onClick={savePatientData}
        >
          Save Changes
        </button>
      </div>
      <Fieldset legend="General info">
        <div className="prime-form-line">
          <TextInput
            label="First Name"
            name="firstName"
            value={patient.firstName}
            onChange={onChange}
          />
          <TextInput
            label="Middle Name (optional)"
            name="middleName"
            value={patient.middleName}
            onChange={onChange}
          />
          <TextInput
            label="Last Name"
            name="lastName"
            value={patient.lastName}
            onChange={onChange}
          />
        </div>
        <div className="prime-form-line">
          <TextInput
            label="Lookup ID"
            name="lookupId"
            value={patient.lookupId}
            onChange={onChange}
          />
          <Dropdown
            label="Role"
            name="role"
            selectedValue={patient.role}
            onChange={onChange}
            options={[
              { label: "Staff", value: "staff" },
              { label: "Resident", value: "resident" },
              { label: "Student", value: "student" },
              { label: "Visitor", value: "visitor" },
            ]}
          />
        </div>
        <div className="prime-form-line">
          <TextInput
            label="Date of Birth (MM/dd/yyyy)"
            name="birthDate"
            value={patient.birthDate}
            onChange={onChange}
          />
        </div>
      </Fieldset>
      <Fieldset legend="Contact Information">
        <div className="prime-form-line">
          <TextInput
            label="Phone Number"
            name="telephone"
            value={patient.telephone}
            addClass="prime-phone"
            onChange={onChange}
          />
          <TextInput
            label="Email Address"
            name="email"
            value={patient.email}
            addClass="prime-email"
            onChange={onChange}
          />
        </div>
        <div className="prime-form-line">
          <TextInput
            label="Street address 1"
            name="street"
            value={patient.street}
            addClass="prime-street-address"
            onChange={onChange}
          />
        </div>
        <div className="prime-form-line">
          <TextInput
            label="Street address 2"
            name="streetTwo"
            value={patient.streetTwo}
            addClass="prime-street-address"
            onChange={onChange}
          />
        </div>
        <div className="prime-form-line">
          <TextInput
            label="City"
            name="city"
            value={patient.city}
            addClass="prime-city"
            onChange={onChange}
          />
          <TextInput
            label="County"
            name="county"
            value={patient.county}
            addClass="prime-county"
            onChange={onChange}
          />
          <Dropdown
            label="State"
            name="state"
            selectedValue={patient.state}
            options={stateCodes.map((c) => ({ label: c, value: c }))}
            addClass="prime-state"
            onChange={onChange}
          />
          <TextInput
            label="Zip"
            name="zipCode"
            value={patient.zipCode}
            addClass="prime-zip"
            onChange={onChange}
          />
        </div>
      </Fieldset>
      <Fieldset legend="Demographics">
        <div>
          <Checkboxes
            legend="Race"
            displayLegend
            name="race"
            checkedValues={patient.race}
            checkboxes={[
              {
                value: "native",
                label: "American Indian or Alaskan Native",
              },
              {
                value: "asian",
                label: "Asian",
              },
              {
                value: "black",
                label: "Black or African American",
              },
              {
                value: "pacific",
                label: "Native Hawaiian or other Pacific Islander",
              },
              {
                value: "white",
                label: "White",
              },
              {
                value: "unknown",
                label: "Unknown",
              },
              {
                value: "refused",
                label: "Refused to Answer",
              },
            ]}
            onChange={onChange}
          />
        </div>
        <div>
          <RadioGroup
            legend="Ethnicity"
            displayLegend
            name="ethnicity"
            buttons={[
              { label: "Hispanic or Latino", value: "hispanic" },
              { label: "Not Hispanic", value: "not_hispanic" },
            ]}
            selectedRadio={patient.ethnicity}
            onChange={onChange}
          />
        </div>
        <div>
          <RadioGroup
            legend="Biological Sex"
            displayLegend
            name="gender"
            buttons={[
              { label: "Male", value: "male" },
              { label: "Female", value: "female" },
              { label: "Other", value: "other" },
            ]}
            selectedRadio={patient.gender}
            onChange={onChange}
          />
        </div>
      </Fieldset>
      <Fieldset legend="Other">
        <div>
          <RadioGroup
            legend="Resident in congregate care/living setting?"
            displayLegend
            name="residentCongregateSetting"
            buttons={[
              { label: "Yes", value: "YES" },
              { label: "No", value: "NO" },
            ]}
            selectedRadio={patient.residentCongregateSetting}
            onChange={onChange}
          />
        </div>
        <div>
          <RadioGroup
            legend="Work in Healthcare?"
            displayLegend
            name="employedInHealthcare"
            buttons={[
              { label: "Yes", value: "YES" },
              { label: "No", value: "NO" },
            ]}
            selectedRadio={patient.employedInHealthcare}
            onChange={onChange}
          />
        </div>
      </Fieldset>
      <Fieldset legend="Test History">
        {patient.testResults && patient.testResults.length && (
          <table className="usa-table usa-table--borderless">
            <thead>
              <tr>
                <th scope="col">Date of Test</th>
                <th scope="col">Result</th>
              </tr>
            </thead>
            <tbody>
              {patient.testResults.map((r, i) => (
                <tr key={i}>
                  <td>{moment(r.dateTested).format("MMM DD YYYY")}</td>
                  <td>{r.result}</td>
                </tr>
              ))}
            </tbody>
          </table>
        )}
      </Fieldset>
      <div className="prime-edit-patient-heading">
        <div></div>
        <button
          className="usa-button prime-save-patient-changes"
          disabled={!formChanged}
          onClick={savePatientData}
        >
          Save Changes
        </button>
      </div>
    </main>
  );
};

export default PatientForm;
