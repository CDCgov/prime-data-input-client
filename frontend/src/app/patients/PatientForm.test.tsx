import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import PatientForm from "./PatientForm";

const mockStore = configureStore([]);

jest.mock("react-router-dom", () => ({
  Prompt: (props: any) => <></>,
  useHistory: () => ({
    listen: jest.fn(),
    push: jest.fn(),
  }),
}));

jest.mock("@microsoft/applicationinsights-react-js", () => ({
  useAppInsightsContext: () => {},
  useTrackEvent: jest.fn(),
}));

describe("PatientForm", () => {
  it("snapshot", () => {
    const store = mockStore({
      patient: {
        internalId: "51c73542-dc64-4ee8-8368-926fcc61b39e",
        residentCongregateSetting: true,
        employedInHealthcare: true,
        birthDate: "1987-01-31",
      },
      plid: "foo",
      facility: {
        id: "123",
      },
      facilities: [],
    });
    const patient = {
      internalId: "51c73542-dc64-4ee8-8368-926fcc61b39e",
      firstName: "Malik",
      lastName: "Massey",
      middleName: "McKenzie Albert",
      birthDate: "1987-01-31",
      residentCongregateSetting: true,
      employedInHealthcare: true,
      gender: null,
    };
    const facility = {
      id: "123",
    };
    const component = renderer.create(
      <Provider store={store}>
        <MockedProvider mocks={[]} addTypename={false}>
          <PatientForm
            patient={patient}
            activeFacilityId={facility.id}
            patientId={patient.internalId}
            isPxpView={true}
            backCallback={() => {}}
            saveCallback={() => {}}
          />
        </MockedProvider>
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
