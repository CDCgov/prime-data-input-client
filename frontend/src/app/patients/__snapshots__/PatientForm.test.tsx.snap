// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientForm snapshot 1`] = `
<main
  className="prime-edit-patient prime-home padding-top-0"
>
  <div
    className=""
  >
    <p>
      All fields marked with 
      <span
        className="usa-sr-only"
      >
        Required
      </span>
      <span
        className="usa-hint--required"
      >
         *
      </span>
       are required
    </p>
    <div
      className="prime-formgroup"
    >
      <fieldset
        className="usa-fieldset"
      >
        <legend
          className="prime-formgroup-heading usa-legend"
        >
          General info
        </legend>
        <div
          className="usa-form"
        >
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              First name
              <span
                className="usa-sr-only"
              >
                Required
              </span>
              <span
                className="usa-hint--required"
              >
                 *
              </span>
            </label>
            <input
              aria-required={true}
              className="usa-input"
              id="text-test-1"
              name="firstName"
              onChange={[Function]}
              type="text"
              value="Malik"
            />
          </div>
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Middle name
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <input
              aria-required="false"
              className="usa-input"
              id="text-test-1"
              name="middleName"
              onChange={[Function]}
              type="text"
              value="McKenzie Albert"
            />
          </div>
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Last name
              <span
                className="usa-sr-only"
              >
                Required
              </span>
              <span
                className="usa-hint--required"
              >
                 *
              </span>
            </label>
            <input
              aria-required={true}
              className="usa-input"
              id="text-test-1"
              name="lastName"
              onChange={[Function]}
              type="text"
              value="Massey"
            />
          </div>
        </div>
        <div
          className="usa-form"
        >
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Lookup ID
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <input
              aria-required="false"
              className="usa-input"
              id="text-test-1"
              name="lookupId"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="usa-form-group prime-dropdown "
          >
            <label
              className="usa-label"
              htmlFor="drop-test-1"
            >
              Role
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <select
              className="usa-select"
              id="drop-test-1"
              name="role"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                -Select-
              </option>
              <option
                value="STAFF"
              >
                Staff
              </option>
              <option
                value="RESIDENT"
              >
                Resident
              </option>
              <option
                value="STUDENT"
              >
                Student
              </option>
              <option
                value="VISITOR"
              >
                Visitor
              </option>
            </select>
          </div>
        </div>
        <div
          className="usa-form"
        >
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Date of birth (mm/dd/yyyy)
              <span
                className="usa-sr-only"
              >
                Required
              </span>
              <span
                className="usa-hint--required"
              >
                 *
              </span>
            </label>
            <input
              aria-required={true}
              className="usa-input"
              id="text-test-1"
              name="birthDate"
              onChange={[Function]}
              type="date"
              value="1987-01-31"
            />
          </div>
        </div>
      </fieldset>
    </div>
    <div
      className="prime-formgroup"
    >
      <fieldset
        className="usa-fieldset"
      >
        <legend
          className="prime-formgroup-heading usa-legend"
        >
          Contact information
        </legend>
        <div
          className="usa-form"
        >
          <div
            className="grid-row grid-gap"
          >
            <div
              className="mobile-lg:grid-col-6"
            >
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="text-test-1"
                >
                  Phone number
                  <span
                    className="usa-sr-only"
                  >
                    Required
                  </span>
                  <span
                    className="usa-hint--required"
                  >
                     *
                  </span>
                </label>
                <input
                  aria-required={true}
                  className="usa-input"
                  id="text-test-1"
                  name="telephone"
                  onChange={[Function]}
                  type="tel"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Email address
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <input
              aria-required="false"
              className="usa-input"
              id="text-test-1"
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className="usa-form"
        >
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Street address 1
              <span
                className="usa-sr-only"
              >
                Required
              </span>
              <span
                className="usa-hint--required"
              >
                 *
              </span>
            </label>
            <input
              aria-required={true}
              className="usa-input"
              id="text-test-1"
              name="street"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="usa-form"
        >
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Street address 2
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <input
              aria-required="false"
              className="usa-input"
              id="text-test-1"
              name="streetTwo"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="usa-form"
        >
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              City
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <input
              aria-required="false"
              className="usa-input"
              id="text-test-1"
              name="city"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              County
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <input
              aria-required="false"
              className="usa-input"
              id="text-test-1"
              name="county"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="grid-row grid-gap"
          >
            <div
              className="mobile-lg:grid-col-6"
            >
              <div
                className="usa-form-group prime-dropdown "
              >
                <label
                  className="usa-label"
                  htmlFor="drop-test-1"
                >
                  State
                  <span
                    className="usa-sr-only"
                  >
                    Required
                  </span>
                  <span
                    className="usa-hint--required"
                  >
                     *
                  </span>
                </label>
                <select
                  className="usa-select"
                  id="drop-test-1"
                  name="state"
                  onChange={[Function]}
                  value=""
                >
                  <option>
                    - Select -
                  </option>
                  <option
                    value="AK"
                  >
                    AK
                  </option>
                  <option
                    value="AL"
                  >
                    AL
                  </option>
                  <option
                    value="AR"
                  >
                    AR
                  </option>
                  <option
                    value="AS"
                  >
                    AS
                  </option>
                  <option
                    value="AZ"
                  >
                    AZ
                  </option>
                  <option
                    value="CA"
                  >
                    CA
                  </option>
                  <option
                    value="CO"
                  >
                    CO
                  </option>
                  <option
                    value="CT"
                  >
                    CT
                  </option>
                  <option
                    value="DC"
                  >
                    DC
                  </option>
                  <option
                    value="DE"
                  >
                    DE
                  </option>
                  <option
                    value="FL"
                  >
                    FL
                  </option>
                  <option
                    value="GA"
                  >
                    GA
                  </option>
                  <option
                    value="GU"
                  >
                    GU
                  </option>
                  <option
                    value="HI"
                  >
                    HI
                  </option>
                  <option
                    value="IA"
                  >
                    IA
                  </option>
                  <option
                    value="ID"
                  >
                    ID
                  </option>
                  <option
                    value="IL"
                  >
                    IL
                  </option>
                  <option
                    value="IN"
                  >
                    IN
                  </option>
                  <option
                    value="KS"
                  >
                    KS
                  </option>
                  <option
                    value="KY"
                  >
                    KY
                  </option>
                  <option
                    value="LA"
                  >
                    LA
                  </option>
                  <option
                    value="MA"
                  >
                    MA
                  </option>
                  <option
                    value="MD"
                  >
                    MD
                  </option>
                  <option
                    value="ME"
                  >
                    ME
                  </option>
                  <option
                    value="MI"
                  >
                    MI
                  </option>
                  <option
                    value="MN"
                  >
                    MN
                  </option>
                  <option
                    value="MO"
                  >
                    MO
                  </option>
                  <option
                    value="MS"
                  >
                    MS
                  </option>
                  <option
                    value="MT"
                  >
                    MT
                  </option>
                  <option
                    value="NC"
                  >
                    NC
                  </option>
                  <option
                    value="ND"
                  >
                    ND
                  </option>
                  <option
                    value="NE"
                  >
                    NE
                  </option>
                  <option
                    value="NH"
                  >
                    NH
                  </option>
                  <option
                    value="NJ"
                  >
                    NJ
                  </option>
                  <option
                    value="NM"
                  >
                    NM
                  </option>
                  <option
                    value="NV"
                  >
                    NV
                  </option>
                  <option
                    value="NY"
                  >
                    NY
                  </option>
                  <option
                    value="OH"
                  >
                    OH
                  </option>
                  <option
                    value="OK"
                  >
                    OK
                  </option>
                  <option
                    value="OR"
                  >
                    OR
                  </option>
                  <option
                    value="PA"
                  >
                    PA
                  </option>
                  <option
                    value="PR"
                  >
                    PR
                  </option>
                  <option
                    value="RI"
                  >
                    RI
                  </option>
                  <option
                    value="SC"
                  >
                    SC
                  </option>
                  <option
                    value="SD"
                  >
                    SD
                  </option>
                  <option
                    value="TN"
                  >
                    TN
                  </option>
                  <option
                    value="TX"
                  >
                    TX
                  </option>
                  <option
                    value="UT"
                  >
                    UT
                  </option>
                  <option
                    value="VA"
                  >
                    VA
                  </option>
                  <option
                    value="VI"
                  >
                    VI
                  </option>
                  <option
                    value="VT"
                  >
                    VT
                  </option>
                  <option
                    value="WA"
                  >
                    WA
                  </option>
                  <option
                    value="WI"
                  >
                    WI
                  </option>
                  <option
                    value="WV"
                  >
                    WV
                  </option>
                  <option
                    value="WY"
                  >
                    WY
                  </option>
                </select>
              </div>
            </div>
            <div
              className="mobile-lg:grid-col-6"
            >
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="text-test-1"
                >
                  Zip code
                  <span
                    className="usa-sr-only"
                  >
                    Required
                  </span>
                  <span
                    className="usa-hint--required"
                  >
                     *
                  </span>
                </label>
                <input
                  aria-required={true}
                  className="usa-input"
                  id="text-test-1"
                  name="zipCode"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
    <div
      className="prime-formgroup"
    >
      <fieldset
        className="usa-fieldset"
      >
        <legend
          className="prime-formgroup-heading usa-legend"
        >
          Demographics
        </legend>
        <button
          className="usa-button usa-button--unstyled margin-top-1 margin-bottom-2 line-height-sans-2"
          onClick={[Function]}
          type="button"
        >
          Why are we asking for this information?
        </button>
        <fieldset
          className="usa-fieldset prime-radios"
        >
          <legend
            className="usa-legend"
          >
            Race
            <span
              className="usa-hint"
            >
               (optional)
            </span>
          </legend>
          <div
            className="usa-form-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_native_0"
                name="race"
                onChange={[Function]}
                type="radio"
                value="native"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_native_0"
              >
                American Indian or Alaskan Native
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_asian_1"
                name="race"
                onChange={[Function]}
                type="radio"
                value="asian"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_asian_1"
              >
                Asian
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_black_2"
                name="race"
                onChange={[Function]}
                type="radio"
                value="black"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_black_2"
              >
                Black or African American
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_pacific_3"
                name="race"
                onChange={[Function]}
                type="radio"
                value="pacific"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_pacific_3"
              >
                Native Hawaiian or other Pacific Islander
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_white_4"
                name="race"
                onChange={[Function]}
                type="radio"
                value="white"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_white_4"
              >
                White
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_unknown_5"
                name="race"
                onChange={[Function]}
                type="radio"
                value="unknown"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_unknown_5"
              >
                Unknown
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_refused_6"
                name="race"
                onChange={[Function]}
                type="radio"
                value="refused"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_refused_6"
              >
                Refused to answer
              </label>
            </div>
          </div>
        </fieldset>
        <fieldset
          className="usa-fieldset prime-radios"
        >
          <legend
            className="usa-legend"
          >
            Ethnicity
            <span
              className="usa-hint"
            >
               (optional)
            </span>
          </legend>
          <div
            className="usa-form-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_hispanic_0"
                name="ethnicity"
                onChange={[Function]}
                type="radio"
                value="hispanic"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_hispanic_0"
              >
                Hispanic or Latino
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_not_hispanic_1"
                name="ethnicity"
                onChange={[Function]}
                type="radio"
                value="not_hispanic"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_not_hispanic_1"
              >
                Not Hispanic
              </label>
            </div>
          </div>
        </fieldset>
        <fieldset
          className="usa-fieldset prime-radios"
        >
          <legend
            className="usa-legend"
          >
            Biological Sex
            <span
              className="usa-hint"
            >
               (optional)
            </span>
          </legend>
          <div
            className="usa-form-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_male_0"
                name="gender"
                onChange={[Function]}
                type="radio"
                value="male"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_male_0"
              >
                Male
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_female_1"
                name="gender"
                onChange={[Function]}
                type="radio"
                value="female"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_female_1"
              >
                Female
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_other_2"
                name="gender"
                onChange={[Function]}
                type="radio"
                value="other"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_other_2"
              >
                Other
              </label>
            </div>
          </div>
        </fieldset>
      </fieldset>
    </div>
    <div
      className="prime-formgroup"
    >
      <fieldset
        className="usa-fieldset"
      >
        <legend
          className="prime-formgroup-heading usa-legend"
        >
          Other
        </legend>
        <fieldset
          className="usa-fieldset prime-radios"
        >
          <legend
            className="usa-legend"
          >
            Resident in congregate care/living setting?
            <span
              className="usa-sr-only"
            >
              Required
            </span>
            <span
              className="usa-hint--required"
            >
               *
            </span>
          </legend>
          <div
            className="usa-form-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_YES_0"
                name="residentCongregateSetting"
                onChange={[Function]}
                type="radio"
                value="YES"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_YES_0"
              >
                Yes
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_NO_1"
                name="residentCongregateSetting"
                onChange={[Function]}
                type="radio"
                value="NO"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_NO_1"
              >
                No
              </label>
            </div>
          </div>
        </fieldset>
        <fieldset
          className="usa-fieldset prime-radios"
        >
          <legend
            className="usa-legend"
          >
            Work in Healthcare?
            <span
              className="usa-sr-only"
            >
              Required
            </span>
            <span
              className="usa-hint--required"
            >
               *
            </span>
          </legend>
          <div
            className="usa-form-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_YES_0"
                name="employedInHealthcare"
                onChange={[Function]}
                type="radio"
                value="YES"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_YES_0"
              >
                Yes
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className="usa-radio__input"
                disabled={false}
                id="radio-test-1_NO_1"
                name="employedInHealthcare"
                onChange={[Function]}
                type="radio"
                value="NO"
              />
              <label
                className="usa-radio__label"
                htmlFor="radio-test-1_NO_1"
              >
                No
              </label>
            </div>
          </div>
        </fieldset>
      </fieldset>
    </div>
    <div
      className="mobile-lg:display-flex flex-justify-end margin-top-2"
    >
      <button
        className="usa-button usa-button-disabled"
        disabled={true}
        id="edit-patient-save-lower"
        onClick={[Function]}
        type="button"
      >
        Save and continue
      </button>
      <button
        className="usa-button usa-button--outline margin-top-1 mobile-lg:margin-top-0 margin-right-0"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</main>
`;
